### PR TITLE
APP-450: Redesign start date button

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [ ] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/res/drawable/background_rounded_corners_ripple.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
-<ripple android:color="#AAAAAAAA" xmlns:android="http://schemas.android.com/apk/res/android">
+<ripple android:color="?colorControlHighlight" xmlns:android="http://schemas.android.com/apk/res/android">
     <item android:drawable="@drawable/background_rounded_corners"/>
```
Comment (hugokallstrom) [inline app/src/main/res/drawable/background_rounded_corners_ripple.xml:3]: I think what we want here is the [default](https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/res/res/drawable/btn_default_material.xml) ripple color?

File: app/src/main/res/layout/offer_header.xml [hunk 2]
```diff
@@ -73,18 +73,23 @@
...
                 android:padding="@dimen/base_margin">
 
                 <TextView
                     android:id="@+id/startDateLabel"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
-                    android:text="@string/START_DATE_BTN"
+                    android:drawablePadding="@dimen/base_margin"
+                    android:gravity="center_vertical"
+                    android:text="@string/OFFER_ACCESS_DATE"
```
Comment (hugokallstrom) [inline app/src/main/res/layout/offer_header.xml:89]: This text will change depending on if the user is in moving flow or "offer" flow. Logic for that is not the scope for this PR, and is for now hard coded to use moving flow text only.

File: app/src/main/res/layout/offer_header.xml [hunk 3]
```diff
@@ -73,18 +73,23 @@
 
             <androidx.constraintlayout.widget.ConstraintLayout
                 android:id="@+id/startDateContainer"
+                style="?materialButtonTextStyle"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:layout_margin="@dimen/base_margin_double"
-                android:background="@drawable/background_pick_date"
+                android:background="@drawable/background_rounded_corners_ripple"
```
Comment (hedvigoscar) [inline app/src/main/res/layout/offer_header.xml:80]: Given that we are on API 23, we can use `android:foreground="?selectableItemBackground"` 😉

File: app/src/main/res/drawable/ic_calendar.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M17.3571,10.5C16.726,10.5 16.2143,11.0152 16.2143,11.6507V12.8014C16.2143,13.4369 16.726,13.9521 17.3571,13.9521H18.5C19.1312,13.9521 19.6429,13.4369 19.6429,12.8014V11.6507C19.6429,11.0152 19.1312,10.5 18.5,10.5H17.3571Z"
+      android:fillColor="#121212"/>
```
Comment (hedvigoscar) [inline app/src/main/res/drawable/ic_calendar.xml:8]:
```suggestion
      android:fillColor="?colorPrimary"/>
```

File: app/src/main/res/drawable/ic_calendar.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  <path
+      android:pathData="M8.7143,3.2192C8.7143,2.822 8.3945,2.5 8,2.5C7.6055,2.5 7.2857,2.822 7.2857,3.2192V3.4315H4.2857C2.3942,3.4315 1,4.9034 1,6.8121L1,18C1,19.8422 2.4585,21.3806 4.2857,21.3806H19.7143C21.5415,21.3806 23,19.8422 23,18L23,6.8121C23,4.9699 21.5415,3.4315 19.7143,3.4315H16.7143V3.2192C16.7143,2.822 16.3945,2.5 16,2.5C15.6055,2.5 15.2857,2.822 15.2857,3.2192V3.4315H8.7143V3.2192ZM15.2857,5.52V4.8698H8.7143V5.52C8.7143,5.9172 8.3945,6.2392 8,6.2392C7.6055,6.2392 7.2857,5.9172 7.2857,5.52V4.8698H4.2857C3.2201,4.8698 2.4286,5.6601 2.4286,6.8121V7.4555H21.5714V6.8121C21.5714,5.7267 20.7157,4.8698 19.7143,4.8698H16.7143V5.52C16.7143,5.9172 16.3945,6.2392 16,6.2392C15.6055,6.2392 15.2857,5.9172 15.2857,5.52ZM2.4286,18L2.4286,8.8938H21.5714L21.5714,18C21.5714,19.0854 20.7156,19.9423 19.7143,19.9423H4.2857C3.2844,19.9423 2.4286,19.0854 2.4286,18Z"
+      android:fillColor="#121212"
```
Comment (hedvigoscar) [inline app/src/main/res/drawable/ic_calendar.xml:11]:
```suggestion
      android:fillColor="?colorPrimary"
```

File: app/src/main/res/drawable/ic_calendar.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (hedvigoscar) [inline app/src/main/res/drawable/ic_calendar.xml:1]: Make sure to run `avocado` on this `VectorDrawable` 🙂

File: app/src/main/res/drawable/background_rounded_corners_ripple.xml (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/res/drawable/background_rounded_corners_ripple.xml:3]: We might be able to deprecate this drawable given that we are on API 23, check my other comment 🙂

File: app/src/main/res/layout/offer_header.xml (hunk 3 above)
Comment (hugokallstrom) [reply app/src/main/res/layout/offer_header.xml:80]: I want rounded corners though, with foreground the whole layout is targeted (which is squared)

File: app/src/main/res/drawable/background_rounded_corners_ripple.xml (hunk 1 above)
Comment (hugokallstrom) [reply app/src/main/res/drawable/background_rounded_corners_ripple.xml:3]: Ah good catch

File: app/src/main/res/layout/offer_header.xml (hunk 3 above)
Comment (hedvigoscar) [reply app/src/main/res/layout/offer_header.xml:80]: Sensible!

Comment (hedvigoscar) [reply app/src/main/res/layout/offer_header.xml:80]: `?selectableItemBackgroundBorderless` is rounded though, although it's not a squircle, to be fair. You can check it out if you want, but it's not super-important.

**Review by hedvigoscar — APPROVED** [review]

**Pull request merged**